src/utils/common.js

```diff
@@ -621,3 +621,6 @@ export const valueForAddress = <V>(
 export const setValueForAddress = <V>(record: Record<V>, address: string, value: V) => {
   record[addressAsKey(address)] = value;
 };
+
+
+export const parseDate = (date: Date | string | number): Date => new Date(date);
```

Comment:
I'm wondering if we need date parsing function if all it does is `new Date(date)`?

Reply:
Maybe we should return `null` for cases where input is null or when the parsed date is invalid (e.g. incorrect string value). Pls compare: https://stackoverflow.com/a/1353711/484499
Also for undefined input we will return the current date, which might not be the expected value.
Also consider using parsing function from `date-fns` which we are already including.

Reply:
Removed `parseDate` and replaced either with `new Date` or newly added `parseTimestamp` where needed.